src/expint.jl

```diff
@@ -208,9 +208,10 @@ En_safe_gamma_term(ν::Integer, z::Real) = (z ≥ 0 || isodd(ν) ? 1 : -1) * exp
 # returns the two terms from the above equation separately
 function En_cf_gamma(ν::Number, z::Number, n::Int=1000)
     A = float(1 - ν)
-    B::typeof(A) = 1
-    Bprev::typeof(A) = 0
-    Aprev::typeof(A) = 1
+    A, _ = promote(A, z)
+    B = one(A)
```

Comment:
`one(A)` is not necessarily of the same type as `A`. If you want that both arguments are of the same type, you could use e.g.
```suggestion
    B = oneunit(A)
```